### PR TITLE
Avoid starting config flow and show alert dialog early if single config entry only

### PR DESCRIPTION
**Pull request by jpbede**
## Proposed change
Only show the "Add entry" button if the integration signals that it supports multiple entries, otherwise show the user a info that the integration only allows one entry; instead of starting a flow, and show the user that there is already an entry and that they can only configure one.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:
- Link to core pull request: https://github.com/home-assistant/core/pull/109505

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by jpbede** [comment]
Agreed, I re-added the button as disabled with the info below. Or should we handled this just with a dialog?

image

File: src/panels/config/integrations/dialog-single-config-entry-only.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import "@material/mwc-button/mwc-button";
+import { css, CSSResultGroup, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { HomeAssistant } from "../../../types";
+import { SingleConfigEntryOnlyDialogParams } from "./show-add-integration-dialog";
+
+@customElement("dialog-single-config-entry-only")
```
Comment (bramkragten) [inline src/panels/config/integrations/dialog-single-config-entry-only.ts:8]: Why did you create a whole new dialog for this? We can just use an alert dialog?

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 2]
```diff
@@ -513,6 +519,13 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
                         `ui.panel.config.integrations.integration_page.add_entry`
                       )}
                 </ha-button>
+                ${single_config_entry_only
+                  ? html`<ha-alert alert-type="info"
+                      >${this.hass.localize(
+                        `ui.panel.config.integrations.integration_page.single_config_entry_only`
+                      )}</ha-alert
+                    >`
+                  : ``}
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:528]:
```suggestion
                  : nothing}
```

File: src/panels/config/integrations/dialog-single-config-entry-only.ts (hunk 1 above)
Comment (jpbede) [reply src/panels/config/integrations/dialog-single-config-entry-only.ts:8]: No specific reason, just saw how we did it with the YAML dialog. Will change this to the alert dialog later.

**Comment by elupus** [comment]
Want there some cases where re setting up an integration would update the config on the old entry? It was the only way to retain history for entities.

**Comment by jpbede** [comment]
This is still possible, just don't set the manifest option. The option is meant for integrations like `sun`, which immediately abort if they already have a config entry.

**Comment by balloob** [comment]
> I believe that the buttons/UX elements should always be there, but be either disabled or provide an error message with an explanation on click (or maybe even a combination of those?)

Correct. UX guidelines say we should always show the button and not disabled. Clicking on it will show the error. This means that it would be the same behavior as it was before the manifest option. 

I also think that showing a permanent warning is overkill. Most of the times people come to this page to do something else, and now their attention is always drawn to this option.

I would suggest that we limit the impact of this PR to update the types.

I think that the feature itself can still be beneficial, like limiting discovery.

**Comment by jpbede** [comment]
> I would suggest that we limit the impact of this PR to update the types.

I've now restored the old behavior. But we need to at least add the alert dialogs, otherwise the user won't get any information if we use the new manifest option. Core no longer starts a flow if an entry already exists and the option is used.

File: src/translations/en.json [hunk 3]
```diff
@@ -3978,7 +3978,9 @@
             "missing_matter": "To add a {brand} device, you first need the {integration} integration and {supported_hardware_link}. Do you want to proceed with the setup of {integration}?",
             "matter_mobile_app": "You need to use the Home Assistant Companion app on your mobile phone to commission Matter devices.",
             "supported_hardware": "supported hardware",
-            "proceed": "Proceed"
+            "proceed": "Proceed",
+            "single_config_entry_title": "This integration allows only one configuration",
+            "single_config_entry": "The {integration_name} supports only one configuration. Adding additional ones is not needed."
```
Comment (bramkragten) [inline src/translations/en.json:3983]:
```suggestion
            "single_config_entry": "{integration_name} supports only one configuration. Adding additional ones is not needed."
```

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**